src/jdk.compiler/share/classes/com/sun/source/tree/EnhancedForLoopTree.java

```diff
@@ -41,12 +43,37 @@
  * @since 1.6
  */
 public interface EnhancedForLoopTree extends StatementTree {
+    /**
+     * "Enhanced" {@code for} declarations come in two forms:
+     * <ul>
+     * <li> local variable declarations and
+     * <li> record patterns
+     * </ul>
+     *
+     * @since 20
+     */
+    @PreviewFeature(feature=PreviewFeature.Feature.RECORD_PATTERNS, reflective=true)
+    public enum DeclarationKind {
+        /** enum constant for local variable declarations */
+        VARIABLE,
+        /** enum constant for record pattern declarations */
+        PATTERN
+    }
+
     /**
      * Returns the control variable for the loop.
-     * @return the control variable
+     * @return the control variable, or {@code null} if this "enhanced" {@code for} uses a pattern
```

Comment:
nit: this javadoc line is probably too long, usually we try to respect the 80 columns max width. Not sure why but I have myself got this comment in the past